lib/data/schema.js

```diff
@@ -215,7 +215,22 @@ const expectedFormAttachments = (xml) => {
     // if we have an <input query= at all we expect an itemsets.csv file.
     const itemsets = hasItemsets.orElse(false) ? [{ type: 'file', name: 'itemsets.csv' }] : [];
 
-    return externalInstances.concat(mediaFiles).concat(itemsets);
+    // now we need to deduplicate our expected files. we consider type and name
+    // a composite key, and we only add such pairs once. but duplicate names with
+    // different types will be sent through, as they are different expectations.
+    // n.b. this function returns the /expected/ list of files. so removing a dupe:
+    // 1. fulfills the contract: that one file we expect to see once.
+    // 2. /allows/ the duplicated file as one entry: the database constraint will
+    //    eventually complain about duplicate filenames.
+    const seen = {};
+    const result = [];
+    for (const attachment of externalInstances.concat(mediaFiles).concat(itemsets)) {
+      if ((seen[attachment.name] == null) || (seen[attachment.name] != attachment.type))
```

Comment:
I don't think the difference matters in this case, but I suspect the linter will ask for `!==` over `!=`.

Reply:
oh, probably